ams/server/src/main/java/com/netease/arctic/server/catalog/InternalCatalog.java

```diff
@@ -95,7 +117,12 @@ public List<TableIDWithFormat> listTables() {
         .collect(Collectors.toList());
   }
 
-  public ServerTableIdentifier createTable(TableMetadata tableMetadata) {
+  public abstract <A> InternalTableCreator newTableCreator(
```

Comment:
I noticed that only fixed types of parameters are accepted here. Should we consider using templates?